pywren/wren.py

```diff
@@ -556,28 +557,36 @@ def result(self, timeout=None, check_only=False, throw_except=True):
 
         self._call_invoker_result = call_invoker_result
 
+        self.run_status = call_status # this is the remote status information
+        self.invoke_status = self._invoke_metadata # local status information
+
+
         if call_success:
 
             self._return_val = call_invoker_result['result']
             self._state = JobState.success
-        else:
+            return self._return_val
+
+        elif call_success == False and throw_except:
```

Comment:
`call_success` is always False here - so this can be `if throw_except` (since you return from previous if)